airbyte-integrations/connectors/source-mailchimp/source_mailchimp/manifest.yaml

```diff
@@ -0,0 +1,366 @@
+
+  retriever:
+    type: SimpleRetriever
+    record_selector:
+      type: RecordSelector
+      extractor:
+        type: DpathExtractor
+        field_path: ["{{ parameters.get('data_field') }}"]
+    paginator:
+      type: "DefaultPaginator"
+      pagination_strategy:
+        type: "OffsetIncrement"
+        page_size: 1000
+      page_size_option:
+        type: "RequestOption"
+        inject_into: "request_parameter"
+        field_name: "page_size"
+      page_token_option:
+        type: "RequestOption"
+        inject_into: "request_parameter"
+        field_name: "offset"
+    requester:
+      type: CustomRequester
+      class_name: source_mailchimp.components.MailChimpRequester
+      url_base: https://{{ config['data_center'] }}.api.mailchimp.com/3.0/
+      http_method: GET
+      authenticator:
+        type: SelectiveAuthenticator
+        authenticator_selection_path: ["credentials", "auth_type"]
+        authenticators:
+          oauth2.0: "#/definitions/bearer_authenticator"
+          apikey: "#/definitions/basic_authenticator"
+      request_parameters:
+        exclude_fields: "{{ parameters.get('data_field') }}._links"
+
+  base_stream:
+    retriever:
+      $ref: "#/definitions/retriever"
+
+  base_incremental_stream:
+    retriever:
+      $ref: "#/definitions/retriever"
+      requester:
+        $ref: "#/definitions/retriever/requester"
+        request_parameters:
+          sort_field: "{{ parameters['cursor_field'] }}"
+          sort_dir: "ASC"
+          exclude_fields: "{{ parameters.get('data_field') }}._links"
+    transformations:
+      - "#/definitions/transformer_remove_empty_fields"
+    incremental_sync:
+      type: DatetimeBasedCursor
+      cursor_datetime_formats:
+        - "%Y-%m-%dT%H:%M:%S%z"
+      datetime_format: "%Y-%m-%dT%H:%M:%S.%fZ"
+      cursor_field: "{{ parameters['cursor_field'] }}"
+      start_datetime:
+        type: MinMaxDatetime
+        # TODO: ref to use datetime as optional parameter if possible
+        datetime: "{{ config.get('start_date') or day_delta(-500, '%Y-%m-%dT%H:%M:%S.%fZ') }}"
+        datetime_format: "%Y-%m-%dT%H:%M:%S.%fZ"
+      start_time_option:
+        inject_into: request_parameter
+        field_name: "since_{{ parameters['cursor_field'] }}"
+        type: RequestOption
+      end_time_option:
+        inject_into: request_parameter
+        field_name: "before_{{ parameters['cursor_field'] }}"
+        type: RequestOption
+      end_datetime:
+        type: MinMaxDatetime
+        datetime: "{{ now_utc().strftime('%Y-%m-%dT%H:%M:%S.%fZ') }}"
+        datetime_format: "%Y-%m-%dT%H:%M:%S.%fZ"
+      step: P1M
+      cursor_granularity: PT1S
+
+  automations_stream:
+    type: DeclarativeStream
+    $ref: "#/definitions/base_incremental_stream"
+    $parameters:
+      name: "automations"
+      primary_key: "id"
+      path: "automations"
+      data_field: "automations"
+      cursor_field: "create_time"
+
+  campaigns_stream:
```

Comment:
> it feels like a breaking change if empty strings are now

it is not a breaking change, since no schema differences introduced here. I thought that RemoveField is the closest type of transformation that can be applied here.

Reply:
It is not in respect to the schema but it might be in records to the user's consumption. For example, if the user does `select X from Y where address == ""`, his workflow won't work anymore